python/src/pyarrow/adapters/pandas.h

```diff
@@ -36,7 +36,8 @@ namespace pyarrow {
 
 class Status;
 
-Status ArrowToPandas(const std::shared_ptr<arrow::Column>& col, PyObject** out);
+Status ArrowToPandas(const std::shared_ptr<arrow::Column>& col, PyObject** out,
+    PyObject* py_ref);
```

Comment:
`out` should be the last argument, since `py_ref` is an input argument (input args precede output args)